### MTH9866 Modeling and Market Making in Foreign Exchange

#### Homework Assignment 2

#### ShengQuan Zhou
---------------------------

### Question 1

Why are correlations of daily returns of spot vs daily returns of forward prices so high in the FX markets? What are the two requirements a market must support to enforce a high correlation across the forward curve?

#### Answer:

Because FX is a financial market where the two requirements for spot/forward arbitrage hold:
- one can easily store currencies (and receive an interest rate for storing);
- one can easily borrow and short currencies (and pay an interest rate),

traders execute the arbitrage all day, everyday. That arbitrage fixes the market forward price to the fair price
$$
F(t,T) = S(t)e^{\left[R(t,T)-Q(t,T)\right](T-t)}.
$$
The high correlation between daily returns of spot and daily returns of forward prices across the curve arises because the FX markets admit the spot/forward arbitrage that forces the market forward price to equal to the fair forward price. When the spot price moves, all the forwards move accordingly.



### Question 2

Why is risk management more complex for an FX forwards risk manager than for an FX spot risk manager?

#### Answer:

Tenor of the forward contract adds another dimension. Risk management is more complex for FX forwards than for FX spot because not all trades are fungible with each other across the forward curve, like the spot. In the broker market for inter-dealer trades, only benchmark tenors are traded, while clients can trade any settlement date they like. It's unlikely for a client to want to buy exactly the benchmark tenor or the same tenor forward that another client wanted to sell.

### Question 3

Explain why risk to FX forward points can be expressed as risk to non-USD interest rates.

#### Answer:

Based on
$$
F(t,T) = S(t)e^{\left[R(t,T)-Q(t,T)\right](T-t)} \Rightarrow \text{Forward Points} = S(t)\left[e^{\left[R(t,T)-Q(t,T)\right](T-t)}-1\right].
$$
forwards portfolio risk can be decomposed into three components:
- spot risk: hedging risk to spot is handled by spot desk.
- USD interest rates: hedging risk to USD interest rates is handled by rates trading desk, supposedly.
- Non-USD interest rates: the only remaining dimension specific to FX forward points.

### Question 4

Assume a portfolio has just one FX forward position in it, settling on a date $T$ which lies between two benchmark settlement dates $T_1$ and $T_2$. Derive the notionals $N_1$ and $N_2$ of the benchmark forwards which hedge the portfolio risk assuming triangle shocks to the benchmark non-USD interest rates, as shown on page 21 of the lecture notes.

#### Solution:

According to page 9 of the lecture notes, a forward settling in time $T$, receiving 1 unit of asset currency, paying $K$ units of denominated currency, is equivalent to being long 1 unit of a zero coupon bond in asset currency and being short $K$ units of a zero coupon bond in denominated currency, the price of which at spot time $t=0$ is
$$
V(T) = S_0 e^{-Q(T)T} - K e^{-R(T)T},
$$
where $S_0$ is the current spot exchange rate. Similarly,
\begin{eqnarray}
V(T_1) = S_0 e^{-Q(T_1)T_1} - K e^{-R(T_1)T_1},\\
V(T_2) = S_0 e^{-Q(T_2)T_2} - K e^{-R(T_2)T_2}.
\end{eqnarray}
Construct a portfolio that's long one unit of $V(T)$, short $N_1$ units of $V(T_1)$ and $N_2$ units of $V(T_2)$:
$$
\Pi = V(T) - N_1 V(T_1) - N_2 V(T_2),
$$
such that the risk to moving rates $Q(T_1)$ and $Q(T_2)$ is hedged:
\begin{eqnarray}
\frac{d \Pi}{d Q(T_1)} = \frac{\partial \Pi}{\partial Q(T_1)} + \frac{\partial \Pi}{\partial Q(T)}\frac{\partial Q(T)}{\partial Q(T_1)} = S_0 \left[N_1 T_1  e^{-Q(T_1)T_1} - T  e^{-Q(T)T} \frac{\partial Q(T)}{\partial Q(T_1)}\right] =0,\\
\frac{d \Pi}{d Q(T_2)} = \frac{\partial \Pi}{\partial Q(T_2)} + \frac{\partial \Pi}{\partial Q(T)}\frac{\partial Q(T)}{\partial Q(T_2)}=  S_0 \left[N_2 T_2  e^{-Q(T_2)T_2} - T e^{-Q(T)T} \frac{\partial Q(T)}{\partial Q(T_1)}\right] =0.
\end{eqnarray}
Assuming a triangle-shock impact to $Q(T)$,
\begin{eqnarray}
\frac{\partial Q(T)}{\partial Q(T_1)} = \frac{T_2 - T}{T_2-T_1}, \text{ shocked at }T_1\text{ and fixed at }T_2\text{, linear interpolate between }T_2\text{ and }T_1\text{ at }T,\\
\frac{\partial Q(T)}{\partial Q(T_2)} = \frac{T - T_1}{T_2-T_1}, \text{ shocked at }T_2\text{ and fixed at }T_1\text{, linear interpolate between }T_2\text{ and }T_1\text{ at }T,
\end{eqnarray}
we get
\begin{eqnarray}
N_1 T_1  e^{-Q(T_1)T_1} - T  e^{-Q(T)T} \left(\frac{T_2 - T}{T_2-T_1}\right) =0 \Rightarrow N_1 = \left(\frac{T_2 - T}{T_2-T_1}\right) \frac{T}{T_1}  e^{-Q(T)(T-T_1)}, \\
N_2 T_2  e^{-Q(T_2)T_2} - T  e^{-Q(T)T} \left(\frac{T - T_1}{T_2-T_1}\right) =0\Rightarrow N_2 = \left(\frac{T - T_1}{T_2-T_1}\right) \frac{T}{T_2}  e^{-Q(T)(T-T_2)}.
\end{eqnarray}


### Question 5

Explain principal component analysis and factor models, focusing on the differences between the two approaches to reduce dimensionality.

#### Answer:

Principal component analysis and factor models are two different methods for reducing dimensionality in a high-dimensional system.

| Features |  Principle Component Analysis (PCA)   |   Factor Models |
|:------|------|------|
| Parameters     | Non-Parametric |  Parametric, parameterized by $\sigma_1, \sigma_2, \beta_1, \beta_2, \rho$, etc.    |
|General idea|Discovering the main factors driving the moves in the interest rate curve|Fitting historical behavior to a prescribed parametric form of factors|
| Methodology | Calculating the historical covariances of daily changes in interest rates of different tenors, then orthogonalizing the covariance matrix|Solving stochastic differential equation driving interest rates of different tenors, the computing the covariance matrix analytically in terms of the parameters|
| Shocks to curve | Eigenvectors of historical covariance matrix of daily changes in different tenors, representing othogonal shocks to curve | Shocked by a small number of Brownian motion factors in a stochastic differential equation driving different tenors|
|How to reduce dimensionality|Picking off the largest few eigenvalues and approximating all moves in the curves as being linear combinations of the respective eigenvectors|A few parametric factor in the model specification|
|Shape of shocks|Non-parametric output of the calculation|Determined by the model parameters, for example, exponential shocks|
|Strength|Flexibile in terms of shape of the shocks|Robust in evaluating the shocks and easier for traders to understand|
|Weakness|Less robust (historical data artifacts can lead to weird shapes)|Less flexibile in capturing shocks|

### Question 6

This programming question will try to determine whether using a factor-based approach to reducing dimensionality is better than an ad hoc method.

We start by assuming a toy market: spot = 1, asset currency interest rate curve = $Q(T)$ = flat at 3%, and denominated currency interest rate curve = $R(T)$ = flat at 0%. We assume two benchmark dates, $T_1$ = 0.25y and $T_2$ = 1y; we will use forwards to those settlement dates to hedge the forward rate risk (or equivalently, the risk to moves in the asset currency interest rate) of our portfolio.

In the toy market, we assume that we know the dynamics of the asset currency interest rate:
\begin{eqnarray}
&& dQ = \sigma_1 e^{-\beta_1 T}dz_1 + \sigma_2 e^{-\beta_2 T}dz_2,\\
&& \mathbb{E}[dz_1 dz_2] = \rho dt,
\end{eqnarray}
where $\sigma_1 = 1\%/\sqrt{\text{year}}$, $\sigma_2 = 0.8\%/\sqrt{\text{year}}$, $\beta_1 = 0.5/\text{year}$, $\beta_2 = 0.1/\text{year}$, and $\rho = -0.4$. 

The portfolio to hedge has one position: a unit asset-currency notional of a forward contract settling at time $T$. You’ll try this for values of $T$ in [0.1,0.25,0.5,0.75,1,2] to see how performance changes for portfolios with risk to different tenors.

You will try three different hedging strategies: one where you choose the hedge notionals (of forwards settling at times $T_1$ and $T_2$) based on the triangle shock we discussed in class (though as there are only two benchmarks here, the T1 shock will be flat before $T_1$ and the $T_2$ shock will be flat after $T_2$); one where the notionals are set to hedge the actual two shocks from the factors described above; and lastly, one where you don’t hedge at all.

The result should show that setting hedge notionals based on the true factor shocks should provide a better hedge performance than based on the ad hoc triangle shocks. You should analyze just how much better that performance is.

Run a Monte Carlo simulation where you do the following on each run, for each value of $T$, for each of the three hedging strategies described above:


1. Construct a portfolio long 1 unit of the forward settling at time $T$
2. Add in the hedges: two forwards, settling at times $T_1$ and $T_2$, with notionals set to hedge the portfolio (either against the two triangle shocks or against the two factor shocks). Don’t bother adding the hedges in the third hedge scenario where we leave the portfolio unhedged.
3. Simulate the portfolio forward a time $dt=0.001\text{y}$. That will result in the asset-currency rates moving according to the factor model described above, which shocks the benchmark rates for tenors $T_1$ and $T_2$, and for the portfolio’s risk tenor T. Determine the PNL realized. 
Then construct the PNL distributions for the three hedging approaches. The unhedged version is the benchmark: you should compare how much more effectively the PNL standard deviation is reduced by hedging according to the true factors vs hedging according to the ad hoc triangle shocks.

Do this for all the values of T listed above, and discuss your results.

#### Solution:

Similar to the approach to Question 4, construct a portfolio that's long one unit of $V(T)$, short $N_1$ units of $V(T_1)$ and $N_2$ units of $V(T_2)$:
$$
\Pi = V(T) - N_1 V(T_1) - N_2 V(T_2),
$$
such that the risk to moving rates $Q(T_1)$ and $Q(T_2)$ is hedged:
\begin{eqnarray}
\frac{d \Pi}{d Q(T_1)} = \frac{\partial \Pi}{\partial Q(T_1)} + \frac{\partial \Pi}{\partial Q(T)}\frac{\partial Q(T)}{\partial Q(T_1)} = S_0 \left[N_1 T_1  e^{-Q(T_1)T_1} - T  e^{-Q(T)T} \frac{\partial Q(T)}{\partial Q(T_1)}\right] =0 \Rightarrow N_1 = \frac{\partial Q(T)}{\partial Q(T_1)} \frac{T}{T_1}  e^{-Q(T)(T-T_1)}, \\
\frac{d \Pi}{d Q(T_2)} = \frac{\partial \Pi}{\partial Q(T_2)} + \frac{\partial \Pi}{\partial Q(T)}\frac{\partial Q(T)}{\partial Q(T_2)}=  S_0 \left[N_2 T_2  e^{-Q(T_2)T_2} - T e^{-Q(T)T} \frac{\partial Q(T)}{\partial Q(T_1)}\right] =0\Rightarrow N_2 = \frac{\partial Q(T)}{\partial Q(T_2)} \frac{T}{T_2}  e^{-Q(T)(T-T_2)},
\end{eqnarray}
where the partial derivative $\frac{\partial Q(T)}{\partial Q(T_1)}$ is taken with $Q(T_2)$ held constant, and vice versa for $\frac{\partial Q(T)}{\partial Q(T_2)}$. 

-------------------------

For triangular shocks, the explicit expressions for the partial derivatives mentioned above were derived in Question 4:
\begin{eqnarray}
\frac{\partial Q(T)}{\partial Q(T_1)} = \frac{T_2-T}{T_2-T_1},\\
\frac{\partial Q(T)}{\partial Q(T_2)} = \frac{T-T_1}{T_2-T_1}.
\end{eqnarray}

-------------------------

For shocks from factors,
\begin{eqnarray}
dQ(T_1) = \sigma_1 e^{-\beta_1 T_1}dz_1 + \sigma_2 e^{-\beta_2 T_1}dz_2,\\
dQ(T_2) = \sigma_1 e^{-\beta_1 T_2}dz_1 + \sigma_2 e^{-\beta_2 T_2}dz_2.
\end{eqnarray}
For $Q(T_1)$ shocked and $Q(T_2)$ fixed,
$$
dQ(T_2) = \sigma_1 e^{-\beta_1 T_2}dz_1 + \sigma_2 e^{-\beta_2 T_2}dz_2 = 0 \Rightarrow \frac{dz_1}{dz_2} = -\frac{\sigma_2 e^{-\beta_2 T_2}}{\sigma_1 e^{-\beta_1 T_2}}.
$$
Substitute back to $dQ(T_1) = \sigma_1 e^{-\beta_1 T_1}dz_1 + \sigma_2 e^{-\beta_2 T_1}dz_2$, we get
\begin{eqnarray}
\sigma_1dz_1 &=& \frac{e^{-\beta_2 T_2}}{e^{-\beta_1 T_1  - \beta_2 T_2 } - e^{-\beta_1 T_2  - \beta_2 T_1  }}dQ(T_1),\\
\sigma_2dz_2 &=& -\frac{e^{-\beta_1 T_2}}{e^{-\beta_1 T_1  - \beta_2 T_2 } - e^{-\beta_1 T_2  - \beta_2 T_1  }}dQ(T_1).
\end{eqnarray}
Thus,
$$
dQ(T) = \sigma_1 e^{-\beta_1 T}dz_1 + \sigma_2 e^{-\beta_2 T}dz_2
=  \frac{e^{-\beta_2 T_2 - \beta_1 T}-e^{-\beta_1 T_2 - \beta_2 T}}{e^{-\beta_1 T_1  - \beta_2 T_2 } - e^{-\beta_1 T_2  - \beta_2 T_1  }}  dQ(T_1) \Rightarrow \frac{\partial Q(T)}{\partial Q(T_1)} = \frac{e^{-\beta_2 T_2 - \beta_1 T}-e^{-\beta_1 T_2 - \beta_2 T}}{e^{-\beta_1 T_1  - \beta_2 T_2 } - e^{-\beta_1 T_2  - \beta_2 T_1  }}.
$$
Similarly, for $Q(T_2)$ shocked and $Q(T_1)$ fixed,
$$
\frac{\partial Q(T)}{\partial Q(T_2)} = \frac{e^{-\beta_1 T_1 - \beta_2 T}-e^{-\beta_2 T_1 - \beta_1 T}}{e^{-\beta_1 T_1  - \beta_2 T_2  } - e^{-\beta_1 T_2  - \beta_2 T_1 } }.
$$

-------------------------

We briefly tabulate the results below,

|  |  Trianglar shock  |   Factor Model shock |
|:------|------|------|
|$\frac{\partial Q(T)}{\partial Q(T_1)}$|$\frac{T_2-\max{(T,T_1)}}{T_2-T_1}$|$\frac{e^{-\beta_2 T_2 - \beta_1 T}-e^{-\beta_1 T_2 - \beta_2 T}}{e^{-\beta_1 T_1  - \beta_2 T_2 } - e^{-\beta_1 T_2  - \beta_2 T_1  }}$|
|$\frac{\partial Q(T)}{\partial Q(T_2)}$|$\frac{\min{(T,T_2)}-T_1}{T_2-T_1}$|$\frac{e^{-\beta_1 T_1 - \beta_2 T}-e^{-\beta_2 T_1 - \beta_1 T}}{e^{-\beta_1 T_1  - \beta_2 T_2  } - e^{-\beta_1 T_2  - \beta_2 T_1 } }$|

Note that, interestingly, the factor model shock is reduced to the triangular shock in the limit $\beta_1,\beta_2\rightarrow 0$, i.e. the limit of weakly decaying shocks:
\begin{eqnarray}
\frac{e^{-\beta_2 T_2 - \beta_1 T}-e^{-\beta_1 T_2 - \beta_2 T}}{e^{-\beta_1 T_1  - \beta_2 T_2 } - e^{-\beta_1 T_2  - \beta_2 T_1  }} \rightarrow \frac{T_2-T}{T_2-T_1},\\
\frac{e^{-\beta_1 T_1 - \beta_2 T}-e^{-\beta_2 T_1 - \beta_1 T}}{e^{-\beta_1 T_1  - \beta_2 T_2  } - e^{-\beta_1 T_2  - \beta_2 T_1 } }\rightarrow \frac{T-T_1}{T_2-T_1}.
\end{eqnarray}

-------------------------

PnL calculation: for a forward settling in time $T$, receiving 1 unit of asset currency, paying $K$ units of denominated currency, is equivalent to being long 1 unit of a zero coupon bond in asset currency and being short $K$ units of a zero coupon bond in denominated currency, the price of which at spot time $t=0$ is
$$
V(T) = S_0 e^{-Q(T)T} - K e^{-R(T)T},
$$
The change in the price of that forward under a shock $dQ(T)$ is then 
$$
dV(T) = S_0 \left[ e^{-(Q(T)+dQ(T))T} - e^{-Q(T)T} \right] = S_0e^{-Q(T)T} \left[ e^{-dQ(T)T} - 1\right],
$$
independent of the forward strike $K$ or the interest rate $R$ of the denominated currency. The PnL calculation of the portfolio is thus 
$$
dV(T)-N_1 dV(T_1) - N_2 dV(T_2).
$$
Please run the following code to verify.

In [63]:
import math
import scipy
import pandas

In [64]:
class BenchmarkShockSimulator:
    """
        A Monte Carlo simulation of a hedged portfolio:
            1) long an FX forward settling at time T,
            2) short an FX forward settling at time T1,
            3) short an FX forward settling at time T2,
        where asset currency is some non-USD interest rate,
              denominated currency is USD interest rate.
        The rate curve moves according to either
            1) a triangular prescription,
            2) or a parametric 2-factor model.
    """
    
    def __init__(self, sigma1, sigma2, beta1, beta2, rho, T1, T2, Q):
        """
            sigma1: the volatility of the first shock factor
            sigma2: the volatility of the second shock factor
            beta1:  the decaying parameter of the first shock factor
            beta2:  the decaying parameter of the second shock factor
            rho:    the correlation between the two shock factors
            T1:     the time to settlement of the first benchmark forward tenor
            T2:     the time to settlement of the second benchmark forward tenor
            Q:      the initial (background) asset currency forward curve, assumed to be flat
        """
        
        if T1 >= T2:
            raise ValueError('Time order T1<T2 is not imposed!')
        
        self.sigma1 = sigma1
        self.sigma2 = sigma2
        self.beta1 = beta1
        self.beta2 = beta2
        self.rho = rho
        self.T1 = T1
        self.T2 = T2
        self.Q = Q
        
    def run(self, spot, T, dt, nRuns, hedge, seed, debug=False):
        """
            spot:  spot price of asset 
            T:     the settling time of asset-currency notional of a forward contract in portfolio
            dt:    width of a single time step of simulation
            nRuns: number of instantiations of paths
            hedge: 'None' for no hedge; 
                   'Triangle' for hedging by triangular shocks; 
                   'Factor' for hedging by factor models specified for shock evolution.
        """            
                        
        # setup hedging notionals
        if hedge is None:
            N1 = N2 = 0
        elif hedge == 'Triangle':
            if T <= self.T1:
                N1 = 1
                N2 = 0
            elif T >= self.T2:
                N1 = 0
                N2 = 1
            else:
                N1 = (self.T2 - T) / (self.T2 - self.T1)
                N2 = (T - self.T1) / (self.T2 - self.T1)
        elif hedge == 'Factor':
            N1 = (math.exp(-self.beta2 * self.T2 - self.beta1 * T)-math.exp(-self.beta1 * self.T2 - self.beta2 * T))/(math.exp(-self.beta1 * self.T1 - self.beta2 * self.T2)-math.exp(-self.beta1 * self.T2 - self.beta2 * self.T1))
            N2 = (math.exp(-self.beta1 * self.T1 - self.beta2 * T)-math.exp(-self.beta2 * self.T1 - self.beta1 * T))/(math.exp(-self.beta1 * self.T1 - self.beta2 * self.T2)-math.exp(-self.beta1 * self.T2 - self.beta2 * self.T1))
        else:
            raise ValueError('Hedge type must be 1) None 2) Triangle 3) Factor')
        
        N1 *= (T / self.T1) * math.exp(-self.Q * (T - self.T1))
        N2 *= (T / self.T2) * math.exp(-self.Q * (T - self.T2))
        
        # generate instances of shocks
        scipy.random.seed(seed)
        sqrtdt = math.sqrt(dt)
        # independent Brownian shocks
        dw1 = scipy.random.normal(0, sqrtdt, nRuns)
        dw2 = scipy.random.normal(0, sqrtdt, nRuns)
        # correlated Brownian shocks by rho
        dz1 = dw1
        dz2 = self.rho * dw1 + math.sqrt(1 - self.rho * self.rho) * dw2
        
        dQT = self.sigma1 * math.exp(-self.beta1 * T)  * dz1 + self.sigma2 * math.exp(-self.beta2 * T)  * dz2
        dQ1 = self.sigma1 * math.exp(-self.beta1 * self.T1) * dz1 + self.sigma2 * math.exp(-self.beta2 * self.T1) * dz2
        dQ2 = self.sigma1 * math.exp(-self.beta1 * self.T2) * dz1 + self.sigma2 * math.exp(-self.beta2 * self.T2) * dz2
        
        # calculate the portfolio PnLs under the rate shocks
        # Note: in the second exp function, needs to use scipy version to vectorize.
        pnls = spot * math.exp(-self.Q * T) * (scipy.exp(-dQT * T) - 1)
        pnls -= N1 * spot * math.exp(-self.Q * self.T1) * (scipy.exp(-dQ1 * T1) - 1)  
        pnls -= N2 * spot * math.exp(-self.Q * self.T2) * (scipy.exp(-dQ2 * T2) - 1)
        
        return (pnls.mean(), pnls.std())
        

In [65]:
# Static parameters
sigma1 = 0.01
sigma2 = 0.008
beta1 = 0.5
beta2 = 0.1
rho = -0.4
T1 = 0.25
T2 = 1
Q = 0.03

In [66]:
# Create simulator
sim = BenchmarkShockSimulator(sigma1, sigma2, beta1, beta2, rho, T1, T2, Q)

In [67]:
# Simulation parameters
spot = 1
dt = 1e-3
nRuns = 100000
seed = 1
Ts = [0.1, 0.25, 0.5, 0.75, 1, 2]
hedgeTypes = [None,'Triangle','Factor']

In [68]:
df = pandas.DataFrame()
df['T'] = Ts

In [69]:
# Run simulations
for hedgeType in hedgeTypes:
    # choose a nice name for each hedge type
    if hedgeType is None:
        hedgeTypeName = 'No Hedge'
    else:
        hedgeTypeName = hedgeType
        
    # run experiments for different settling times
    stds = []
    for T in Ts:
        mean, std = sim.run(spot, T, dt, nRuns, hedgeType, seed)
        stds.append(std*1e3)
        
    df[hedgeTypeName] = stds

In [70]:
# display the standard deviations in units of 10^-3.
df['Triangle efficiency'] = 1-df['Triangle']/df['No Hedge']
df['Factor efficiency'] = 1-df['Factor']/df['No Hedge']
df.round(6)

T  No Hedge  Triangle    Factor  Triangle efficiency  Factor efficiency
0  0.10  0.030238  0.002042  0.000000             0.932486           0.999988
1  0.25  0.071404  0.000000  0.000000             1.000000           1.000000
2  0.50  0.130707  0.001792  0.000001             0.986290           0.999990
3  0.75  0.180956  0.002558  0.000002             0.985864           0.999990
4  1.00  0.224536  0.000000  0.000000             1.000000           1.000000
5  2.00  0.361658  0.130804  0.000033             0.638321           0.999909

#### Discussions

Hedging by triangular shocks achieves an excellent efficiency (>0.98) for $T_1\le T\le T_2$, reasonable (~0.93) for $T<T_1$, but poorly (~0.64) for $T>T_2$. This means that the simple triangular model is a fairly good model as long as the benchmark tenors used in hedging cover the target range of settling dates in the portfolio. The reason for this performance decline approaching the curve backend is that the assumption of forward curve moving in parallel in the backend (from 1y outward) is unrealistic. In fact, the backend of the forward curve responds to a less extent to the shocks than the front end does. More realistically, the 2y tenor will move less than the 1y tenor. As a consequence, the triangular shock model over-estimates the hedging notional on 1y tenor.

The factor model hedging essentially offsets all movements in $Q(T)$ (>0.9999). This is expected because the shocks are hedged in exactly the same way they are generated, according to the 2-factor stochastic differential equations.